applications/main/nfc/plugins/supported_cards/charliecard.c

```diff
@@ -833,6 +863,9 @@ static uint16_t type_parse(const MfClassicData* data) {
     // bitshift (2bytes = 16 bits) by 6bits for just first 10bits
     const uint16_t type1 = pos_to_num(data, 2, 1, 0, 2) >> 6;
     const uint16_t type2 = pos_to_num(data, 3, 1, 0, 2) >> 6;
+    // might be wise to remove the assertion; then again, it's an effective way
+    // to crowdsource research, as hopefully if this isn't universally true,
+    // someone will come running when their app crashes — probably not a best practice though haha.
     furi_assert(type1 == type2);
```

Comment:
furi_assert() is compiled only when DEBUG=1 is enabled, which it is not by default, nor for production builds in Momentum, you'd need to compile with `./fbt <command> DEBUG=1`, so assert should be fine here! if you want to enforce it in all builds, regardless of DEBUG flag, you can use `furi_check()`

Reply:
Good to know! For my own education: does that mean compiling without DEBUG strips that, and is independent of the debug setting on the flipper? Or is the check kept in the compiled fap regardless, and the DEBUG compile arg simply sets the debug mode flag (locally or globally) on the flipper after flashing/launching?

Reply:
`DEBUG=1` refers to the compiler options, with `DEBUG=0` (default) `furi_assert()` will be stripped completely. the `Debug` option in flipper settings allows to attacha live debugger, and to access some extra hidden functionality (raw reads, debug info in some parsers...), but will not bring back the stripped `furi_assert()` calls